unit-tests/live/d400/test-hdr.py

```diff
@@ -0,0 +1,638 @@
+    ctx = rs.context()
+    device = test.find_first_device_or_exit()
+    if ctx:
+        devices_list = ctx.query_devices()
+        device_count = devices_list.size()
+        if devices_list.size() != 0:
+            depth_sensor = device.first_depth_sensor()
+            if depth_sensor and depth_sensor.supports(rs.option.hdr_enabled):
+                depth_sensor.set_option(rs.option.hdr_enabled, 1)
+                test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+
+                cfg = rs.config()
+                cfg.enable_stream(rs.stream.depth)
+                pipe = rs.pipeline()
+                pipe.start(cfg)
+
+                # initializing the merging filter
+                merging_filter = rs.hdr_merge()
+
+                num_of_iterations_in_serie = 10
+                first_series_last_merged_ts = -1
+                at_least_one_frame_supported_seq_id = False
+                for i in range(0, num_of_iterations_in_serie):
+                    data = pipe.wait_for_frames()
+                    out_depth_frame = data.get_depth_frame()
+
+                    if out_depth_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id):
+                        at_least_one_frame_supported_seq_id = True
+                        # merging the frames from the different HDR sequence IDs
+                        merged_frameset = merging_filter.process(data)
+                        merged_depth_frame = merged_frameset.as_frameset().get_depth_frame()
+
+                        frame_ts = merged_depth_frame.get_frame_metadata(rs.frame_metadata_value.frame_timestamp)
+
+                        if i == (num_of_iterations_in_serie - 1):
+                            first_series_last_merged_ts = frame_ts
+
+                pipe.stop()
+
+                if at_least_one_frame_supported_seq_id:
+                    test.check(first_series_last_merged_ts != -1)
+                    test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+                    pipe.start(cfg)
+
+                    for i in range(0,10):
+                        data = pipe.wait_for_frames()
+                        out_depth_frame = data.get_depth_frame()
+
+                        if out_depth_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id):
+                            # merging the frames from the different HDR sequence IDs
+                            merged_frameset = merging_filter.process(data)
+                            merged_depth_frame = merged_frameset.as_frameset().get_depth_frame()
+
+                            frame_ts = merged_depth_frame.get_frame_metadata(rs.frame_metadata_value.frame_timestamp)
+                            test.check(frame_ts > first_series_last_merged_ts)
+                    pipe.stop()
+                else:
+                    log.e("sequence_id metadata was not supported - skipping test")
```

Comment:
Needed?
Can we support HDR and not sequence_id metadata and pass the test?

Reply:
I don't think so, seems like the merging process is using the 'sequence id' metadata

Reply:
Normally you can replace `is_supported` condition with else with error with
test.check (is supported)

It should be supported :)

Reply:
sure, so I'll replace `at_least_one_frame_supported_seq_id` with `test.check(at_least_one_frame_supported_seq_id)` and remove the condition there